exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/aggregate/StreamingAggBatch.java

```diff
@@ -259,15 +265,16 @@ public IterOutcome innerNext() {
           throw new IllegalStateException(String.format("unknown outcome %s", lastKnownOutcome));
       }
     } else {
-      if ( lastKnownOutcome != NONE && firstBatchForDataSet && !aggregator.isDone()) {
+      if ( lastKnownOutcome != NONE && firstBatchForDataSet && !aggregator.isDone()
+        && aggregator.previousBatchProcessed()) {
```

Comment:
This is the place where `StreamingAgg` after returning previous output batch doesn't know if the incoming batch still has some rows leftover to process before calling next(). The new added method `previousBatchProcessed` will provide that info now.

Reply:
done